sdks/python/apache_beam/transforms/stats_test.py

```diff
@@ -482,13 +482,74 @@ def test_alternate_quantiles(self):
           equal_to([["ccccc", "aaa", "b"]]),
           label='checkWithKeyAndReversed')
 
+  def test_batched_quantiles(self):
```

Comment:
I have some concerns about test coverage of  ApproximateQuantiles.

1. Do any of the tests exercise merging of accumulators?
2. Do any of the tests exercise collapsing of multiple buffers, including buffers with same & different weights?

Reply:
We should have codecov working again, but somehow retesting didn't trigger it. I think if you add commits to this branch or repush it, codecov will run and may give some signals re: test coverage.

Reply:
1. I think the tests use DirectRunner, so probably no.
2. The approximation will be properly tested only if either the number of inputs will be large with default settings, or max_num_elements and epsilon will be set to extremely low and large values, respectively. I tested approximation with large number of inputs and FlumeCppRunner during development, but it took some time to complete, so it's probably not suitable for continuous testing. It might make sense for me to initialize the CombineFn with the extreme values and test add_input, merge_accumulators and extract_output directly, WDYT?

Reply:
Given that a typo that creeps in might not be discovered until the code runs on a sufficiently large input, I'd be more comfortable merging this if we added targeted unit tests just for methods in question, exercising the logic that is not exercised in direct runner test due to small size/undeterminism.

Reply:
Makes sense. I added a test that creates a combinefn with extremely small max_num_elements and manually split the data to a bunch of accumulators which are then merged. Also, the small max_num_elements will ensure that buffer collapsing and interpolation is exercised.

Reply:
Thank you, much appreciated!